.github/workflows/check-on-push.yml

```diff
@@ -20,10 +20,49 @@ jobs:
       - uses: actions-rs/clippy-check@v1
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all-targets -- -D warnings
         env:
           TAURI_DIST_DIR: ${{ runner.workspace }}/tauri/tauri/examples/communication/dist
           TAURI_DIR: ${{ runner.workspace }}/tauri/tauri/examples/communication/src-tauri
 
+  core_clippy_check:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        feature: [embedded-server, no-server, all-api]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: install webkit2gtk
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y webkit2gtk-4.0
+      - run: rustup component add clippy
+      - name: clippy check
+        working-directory: ./tauri
+        uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all-targets --features ${{ matrix.feature }} -- -D warnings
+        env:
+          TAURI_DIST_DIR: ${{ runner.workspace }}/tauri/tauri/examples/communication/dist
+          TAURI_DIR: ${{ runner.workspace }}/tauri/tauri/examples/communication/src-tauri
+
+  format:
```

Comment:
Since we're limited to a certain number of concurrent jobs, maybe it's better to have `fmt` and `clippy` run as part of the same job?

Reply:
Makes sense, thanks